components/security-mgt/org.wso2.carbon.security.mgt/src/main/java/org/wso2/carbon/security/keystore/dao/impl/PubCertDAOImpl.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.security.keystore.dao.impl;
+
+import org.wso2.carbon.database.utils.jdbc.NamedPreparedStatement;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.carbon.security.keystore.KeyStoreManagementException;
+import org.wso2.carbon.security.keystore.dao.PubCertDAO;
+import org.wso2.carbon.security.keystore.dao.constants.PubCertDAOConstants;
+import org.wso2.carbon.security.keystore.dao.constants.PubCertDAOConstants.PubCertTableColumns;
+import org.wso2.carbon.security.keystore.model.PubCertModel;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * This class provides the implementation of the PubCertDAO interface.
+ */
+public class PubCertDAOImpl implements PubCertDAO {
+
+    private static final String DB_CONN_RETRIEVAL_ERROR_MSG = "Error while getting the DB connection.";
```

Comment:
We can move this error message to a common place

Reply:
Resolved with 062799393567b744e3f873afffe2c87a67b07629